src/execution_plan/ops/op_update.c

```diff
@@ -135,16 +135,18 @@ void _UpdateSchemas(const OpUpdate *op) {
 
         /* Locate node label. */
         AST_GraphEntity* ge = MatchClause_GetEntity(op->ast->matchNode, entityAlias);
-        char *l = ge->label;
+        LabelStoreType t = (ge->t == N_ENTITY) ? STORE_NODE : STORE_EDGE;
+        LabelStore *allStore = GraphContext_AllStore(op->gc, t);
+        LabelStore_UpdateSchema(allStore, 1, &entityProp);
+
         // TODO If the match clause does not provide a label, we must update all the stores
         // affected by the SET clause.
-        LabelStoreType t = (ge->t == N_ENTITY) ? STORE_NODE : STORE_EDGE;
+        char *l = ge->label;
+        if (!l) continue;
```

Comment:
For now this is OK, but I think that in the case the user is updating a node which he did not provide a label for, we need to find out if that node is labeled and update the relevant label store